### DISCUSSION TITLE
Setup for Arduino Nano RP2040 and Arduino Nano ESP32?

### DISCUSSION THREAD

**Discussion by IraSch** [lovyan03/LovyanGFX#621] [Device Configurations]
OK, these are probably stupid questions but I'm not seeing how to configure.

First, in general in all the examples I'm not seeing where to set up the SPI pin settings and more importantly (for devices where the pin settings are defined by hardware), where to designate the CS pin?

Then, which configuration INCLUDE should be used for Arduino Nano RP2040 Connect and which one for Arduino Nano ESP32

Thank you.

**Comment by tobozo** [comment]
in your LGFX config block, SPI pins are declared [at Bus level](https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx_user/LGFX_RP2040_096_Waveshare_sample.hpp#L21-L27), while CS and RST pins are declared [at Panel level](https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx_user/LGFX_RP2040_096_Waveshare_sample.hpp#L33-L35) and TFT_BL is at [Ligh level](https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx_user/LGFX_RP2040_096_Waveshare_sample.hpp#L47-L48)

since neither Arduino Nano RP2040 Connect or Arduino Nano ESP32 come with a pre-soldered display, there's no default pin config for those devices

what you can do is copy that [file](https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx_user/LGFX_RP2040_096_Waveshare_sample.hpp) into your project folder and include it from your sketch, then proceed to edit it

- change the display model if necessary
- edit bus/panel/light settings to fit your wiring and display characteristics
- test `tft.init()`

**Reply by IraSch** [reply]
Thank you. This is what I was missing. 

In addition to changing the pins and display characteristics, do I also change 

    lgfx::Panel_ST7735S _panel_instance;   

to something like this to match my display:

    lgfx::Panel_ILI9488 _panel_instance;

And what are 

      cfg.pin_bl      = 25;
      cfg.pwm_channel = 1;

Interesting thing I noticed (that I should have picked up on before).  I see that     cfg.pin_miso   = -1;    which makes sense since the display never sends to the processor so a MISO pin isn't needed. 

Thanks again.

**Reply by tobozo** [reply]
`cfg.pin_bl` is the backlight pin, you can leave the `cfg.pwm_channel` as is 

` cfg.pin_miso = -1` can happen with some spi-3-wires panels so it's not really strange to see that

for example if your display has no RST pin you can as well assign `cfg.pin_rst = -1`

however if your display has a miso line, you can wire it to any available pin (e.g. pin 13) on the RP2040 and set `cfg.pin_miso = 13`
